src/fsharp/FSharp.Core/control.fs

```diff
@@ -846,6 +847,14 @@ namespace Microsoft.FSharp.Control
         // delayPrim = "bindA (return ()) f"
         let delayA f = callA f ()
 
+        let protectExn (cexn:OperationCanceledException) (edi:ExceptionDispatchInfo) =
+            let exn = edi.GetAssociatedSourceException()
+            let collected =
+                match cexn.InnerException with
+                | null -> [|exn|]
```

Comment:
@saul It is one of the details I habe not completely figured out jet. On one hand you are correct on the other hand it feels wrong that you need to change your exception handling code when another exception is added...

Also I feel like people should not write too much code depending on those inner exceptions. So making things awkward is probably a good thing. Maybe we should add our own now exception type, because then we can add some additional warnings/documentation on its members...

At least I now know why @eiriktsarpalis is so skeptical about this. Even the bcl guys saw that this is hard to do afterwards and maintain compat.

But I still feel like hiding exceptions is not an option (again, bcl guys felt the same and added it in the c# await feature)